### PR TITLE
Review cette PR en priorité Sylvain avant les autres plzz feat: use geolib to calculate distance

### PR DESCRIPTION
**Pull request by Mikazian**
## Description

Create a custom hook to calculate the distance between two users

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [x] Frontend
- [x] Backend
- [ ] Other: .....

## General checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I ran `bun run format`/`bun run lint` before commit
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation where needed
- [ ] My changes generate no new warnings
- [ ] I have checked my code and corrected any misspellings
- [ ] I have updated `develop` and merged to my branch before submitting pull request
- [ ] My pull request generate no conflicts with `develop` branch
- [ ] I requested code review from other team members

## Frontend checklist:

- [ ] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [ ] My `Javascript` generate no new console errors
- [ ] I tested my code cross browsers
- [ ] My slice is pixel perfect for both desktop and mobile according to design
- [ ] I conducted basic QA to assure all features are working
- [ ] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [ ] I tested admin by manually adding content from zero
- [ ] I created easy to use admin experience which is self-explanatory
- [ ] I added description to admin fields in hard-to-understand areas
- [ ] I followed guidelines for naming `Typescript` variables
- [ ] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by ElBarto444 — DISMISSED** [review]

**Review by greegouze — APPROVED** [review]

File: packages/backend/api/src/user/user-crud.ts [hunk 1]
```diff
@@ -11,6 +13,33 @@ interface Request extends ExpressRequest {
...
+        'user.id',
+        'user.name',
+        jsonObjectFrom(
+          eb
+            .selectFrom('city')
+            .select(['city.name', 'city.coordinates'])
+            .whereRef('user.city_id', '=', 'city.id'),
+        ).as('city'),
+      ])
+      .whereRef('user.city_id', '=', 'city.id')
```
Comment (YDrogen) [inline packages/backend/api/src/user/user-crud.ts:33]: Pas besoin de cette ligne normalement, c'est déjà fait via le innerJoin

File: packages/frontend/web/src/hooks/use-distance.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          signal,
+          credentials: 'include',
+        });
+
+        const data = await res.json();
+        setCoordinates(data[0].city.coordinates);
+      };
+
+      fetchDistance().catch((error) => {
+        throw new Error(`Fail to fetch distance: ${String(error)}`);
```
Comment (YDrogen) [inline packages/frontend/web/src/hooks/use-distance.ts:39]: noooon

File: packages/frontend/web/src/hooks/use-distance.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import getDistance from 'geolib/es/getDistance';
+import { useEffect, useState } from 'react';
+
+const API_URL = import.meta.env.VITE_API_URL;
+
+interface CoordinatesBody {
+  coordinates: number[];
+}
+
+export default function useDistance({ ...props }) {
```
Comment (YDrogen) [inline packages/frontend/web/src/hooks/use-distance.ts:10]: Les props sont pas typées 🩴

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/backend/api/src/user/user-crud.ts (hunk 1 above)
Comment (Mikazian) [reply packages/backend/api/src/user/user-crud.ts:33]: corrigé

File: packages/frontend/web/src/hooks/use-distance.ts (hunk 3 above)
Comment (Mikazian) [reply packages/frontend/web/src/hooks/use-distance.ts]: corrigé

File: packages/shared/src/database.ts [hunk 4]
```diff
@@ -53,7 +58,7 @@ export interface HobbyTable {
 export interface CityTable {
   id: Generated<number>;
   name: string;
-  coordinates: Point;
+  coordinates: Point & GeoJson;
```
Comment (greegouze) [inline packages/shared/src/database.ts:61]: Obligé de créer un type GeoJson, pas suffisant avec le type Point, qui est également number, number ?

File: packages/shared/src/index.ts [hunk 5]
```diff
@@ -1,3 +1,43 @@
+import type { GeoJson, Point } from './database';
+
 export * from './database';
 export * from './auth';
 export * from './interaction';
+
+export interface UserBody {
```
Comment (greegouze) [inline packages/shared/src/index.ts:7]: Je l'aurais mis dans un fichier exemple Pref-validations pour garder la cohérence de notre architecture actuelle avec seulement les export dans index

**Review by greegouze — APPROVED** [review]

**Review by ElBarto444 — APPROVED** [review]

File: packages/shared/src/database.ts (hunk 4 above)
Comment (YDrogen) [reply packages/shared/src/database.ts:61]: Bonne question

Comment (Mikazian) [reply packages/shared/src/database.ts:61]: Oui car JsonObjectFrom et JsonArrayFrom pour les type Point, il créé un format GeoJson. Et comme je récupère les coordinates dans un JsonObjectFrom, il convertis les coordinates en GeoJson

File: packages/shared/src/index.ts (hunk 5 above)
Comment (Mikazian) [reply packages/shared/src/index.ts:7]: J'ai créé un fichier user.ts et j'ai déporté mon code.

File: packages/shared/src/database.ts (hunk 4 above)
Comment (Mikazian) [reply packages/shared/src/database.ts:61]: J'ai fais des test, dans un select classique je récupère bien :
`coordinates : { x: number, y: number }`

et dans une JsonObject j'obtiens : 
`coordinates : [ number, number ]`

File: packages/frontend/web/src/hooks/use-distance.ts (hunk 2 above)
Comment (Mikazian) [reply packages/frontend/web/src/hooks/use-distance.ts]: retirer

**Review by greegouze — DISMISSED** [review]

File: packages/shared/src/database.ts (hunk 4 above)
Comment (YDrogen) [reply packages/shared/src/database.ts:61]: Le problème est que le type `GeoJson` est un objet
Du coup niveau type ça donne
image

Comment (Mikazian) [reply packages/shared/src/database.ts:61]: corrigé

**Review by greegouze — DISMISSED** [review]

**Review by greegouze — DISMISSED** [review]

**Review by greegouze — APPROVED** [review]

**Review by greegouze — APPROVED** [review]

**Review by YDrogen — APPROVED** [review]

**Pull request merged**